cypress/integration/core-opensearch-dashboards/opensearch-dashboards/apps/query_enhancements/saved_search.spec.js

```diff
@@ -0,0 +1,513 @@
+const setDataset = (dataset) => {
+  const datasetType = mapDatasetToType(dataset);
+  cy.getElementByTestId('datasetSelectorButton').should('be.visible').click();
+
+  if (datasetType === 'INDEX_PATTERN') {
+    cy.get(`[title="${dataset}"]`).click();
+  } else if (datasetType === 'INDEXES') {
+    cy.getElementByTestId('datasetSelectorAdvancedButton').click();
+    cy.contains('span', 'Indexes').click();
+    cy.contains('span', datasourceName).click();
+    // this element is sometimes being masked by another element
+    cy.contains('span', dataset).should('be.visible').click({ force: true });
+    cy.getElementByTestId('datasetSelectorNext').click();
+
+    cy.getElementByTestId('advancedSelectorTimeFieldSelect').select('timestamp');
+    cy.getElementByTestId('advancedSelectorConfirmButton').click();
+
+    cy.getElementByTestId('datasetSelectorButton').should(
+      'contain.text',
+      `${datasourceName}::${dataset}`
+    );
+  }
+};
+
+const setDatePickerDatesAndSearchIfRelevant = (language) => {
+  if (language === 'OpenSearch SQL') {
+    return;
+  }
+
+  cy.setTopNavDate(START_TIME, END_TIME);
+};
+
+const setSearchConfigurations = ({
+  addFilter,
+  queryString,
+  setHistogramInterval,
+  selectFields,
+  applySort,
+}) => {
+  if (addFilter) {
+    cy.getElementByTestId('showFilterActions').click();
+    cy.getElementByTestId('addFilters').click();
+    cy.getElementByTestId('filterFieldSuggestionList').find('input').type('category');
+    cy.getElementByTestId('comboBoxOptionsList filterFieldSuggestionList-optionsList')
+      .find('button[title="category"]')
+      .click();
+    cy.getElementByTestId('filterOperatorList').find('input').type('is one of');
+    cy.getElementByTestId('comboBoxOptionsList filterOperatorList-optionsList')
+      .find('button[title="is one of"]')
+      .click();
+    cy.getElementByTestId('filterParams').find('input').type('Application');
+    cy.getElementByTestId(
+      'comboBoxOptionsList filterParamsComboBox phrasesParamsComboxBox-optionsList'
+    )
+      .find('button[title="Application"]')
+      .click();
+    // Need to wait here a bit to avoid cypress flakiness
+    cy.wait(750);
+    cy.get('span[title="Application"]').should('be.visible');
+    // Need to wait here a bit to avoid cypress error
+    cy.wait(750);
+    // force is true below because sometimes a dropdown covers the button
+    cy.getElementByTestId('saveFilter').click({ force: true });
+  }
+
+  cy.setQueryEditor(queryString);
+
+  if (setHistogramInterval) {
+    cy.getElementByTestId('discoverIntervalSelect').select('w');
+  }
+
+  if (selectFields) {
+    cy.getElementByTestId('fieldToggle-bytes_transferred').click();
+    cy.getElementByTestId('fieldToggle-personal.name').click();
+
+    // reloading as field filtering doesn't appear right away on cypress. Issue only appears in cypress tests,
+    // so resolving it via a force reload.
+    // cy.reload();
```

Comment:
seems you comment out `cy.reload()`, so the force reload is not needed any more? should we remove line 236 to 238?

Reply:
oops i'll delete